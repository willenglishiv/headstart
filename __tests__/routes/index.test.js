/* eslint-env node, jest */
import request from 'supertest';
import app from '../../app';

describe('Index Route Tests', () => {
  test('Ping route returns Pong', async () => {
    expect.assertions(1);
    const response = await request(app).get('/ping');
    expect(response.text).toBe('Pong');
  });

  test('Status route returns Ok', async () => {
    expect.assertions(1);
    const response = await request(app).get('/status');
    expect(response.text).toBe('Ok');
  });

  test('index route returns rendered HTML', async () => {
    expect.assertions(1);
    const response = await request(app).get('/');
    expect(response.text).toContain('Express');
  });

  test('index route matches snapshot', async () => {
    expect.assertions(1);
    const response = await request(app).get('/');
    expect(response.text).toMatchSnapshot();
  });
});
