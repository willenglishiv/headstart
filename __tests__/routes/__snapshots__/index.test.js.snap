// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Route Tests index route matches snapshot 1`] = `"<!DOCTYPE html><html><head><title>Express</title><link rel=\\"stylesheet\\" href=\\"/stylesheets/style.css\\"></head><body><h1>Express</h1><p>Welcome to Express</p><p>testing a change</p></body></html>"`;
